### PR TITLE
Improve the dock layout and some other UI stuff

### PR DESCRIPTION
**Pull request by mikelow**
## Description
- stack the RawFile list and VGMFile list docks
- add a TitleBar class used with each of the docks
- improve the vertical sizing of docks at launch
- increase tab bar height, and use standard tab bar widths instead of "Expanding" mode on macos
- remove an odd call to setBackgroundRole(QPalette::Dark) after dropping a file into the app

Still trying to feel out a good design for the TitleBar. Some variety in the palette may be beneficial for contrast.

## Motivation and Context
_RawFile VGMFile stack_ - I prefer having the RawFile list be visible alongside the Detected File list, in part because it serves an explanatory purpose, but also because it's useful. The Raw Files list is hard to discover and use as a separate tab. The other issue is that there's no good place to put the tabs; they either clash with the Logs/Collections tabs on the bottom or the table headers on the top.

_Dock sizing_ - Two things here: making sure that Detected Files is given more space than Raw Files, and reducing the height of the Collections List to give the hex view as much space as possible.

_Tab bar height_ - It's just too hard to click right now. Increasing the height brings it closer to the tab bar sizing I see in other apps, like web browsers and IDEs.

## How Has This Been Tested?
Only on MacOS for now. Will definitely be testing on Windows before any merge.

## Screenshots (if appropriate):

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by sykhro** [comment]
> RawFile VGMFile stack - I prefer having the RawFile list be visible alongside the Detected File list, in part because it serves an explanatory purpose, but also because it's useful. The Raw Files list is hard to discover and use as a separate tab. The other issue is that there's no good place to put the tabs; they either clash with the Logs/Collections tabs on the bottom or the table headers on the top.

People have complained about this layout in the past. Personally I find that the nicest setup would be a "tree view", where vgmfiles are contained in rawfiles. It's especially nice if you're looking to dump all raw samples without having to click through a list of different item kinds.

> Dock sizing - Two things here: making sure that Detected Files is given more space than Raw Files, and reducing the height of the Collections List to give the hex view as much space as possible.

It would be nice to make this toggleable. A good chunk of our users are using vgmtrans to convert files and never interact with the hex view.

> Tab bar height - It's just too hard to click right now. Increasing the height brings it closer to the tab bar sizing I see in other apps, like web browsers and IDEs.

Agreed.

**Comment by mikelow** [comment]
> People have complained about this layout in the past. Personally I find that the nicest setup would be a "tree view", where vgmfiles are contained in rawfiles. It's especially nice if you're looking to dump all raw samples without having to click through a list of different item kinds.

What's the complaint? That the Raw File view isn't useful and takes up space? I'm curious how many users are voicing complaints and what they are. 

A tree view could work. As I see it, the upsides are that it consolidates the two views and makes clear which detected files belong to which raw file. The downsides are that it's not as dead-simple a flat table, and it wouldn't be possible to sort across multiple raw files.

To use your example of exporting all samples, I was considering that we add back in a column for file type (sequence, sample collection, instrument set), even if we just use icons to preserve space. To export all samples across multiple files, you just sort by type, easily select all sample collections, and right click. You couldn't do this with a tree view.

If screen real estate is the issue, I'm not sure I find the complaint all that compelling. For one, the raw file view takes up little space. Still, we might consider it spanning the entire left side of the window, and reducing horizontal space for collections. Second, by dragging the dock title bars, it's possible to rearrange the docks however one wants, whether that's making them tabbed or entirely separate windows. In light of this, we might consider making the collections/log views movable via a title bar (perhaps a vertical one?).

At any rate, I don't know what the best solution is. However, I believe that this PR is a major improvement over the status quo.

> It would be nice to make this toggleable. A good chunk of our users are using vgmtrans to convert files and never interact with the hex view.

This I don't agree with. I don't believe the UI is impeding people from this use-case, especially with the movable/resizable docks. Even with the collection view undetachable, a user can make all the views nearly as large as they want.

VGMTrans as an MDI application. Like an IDE, you cannot toggle the MDI area. Trying to add this will result in an app with a confused identity, a confusing design, and little benefit for those who don't use the Hex View.

**Comment by sykhro** [comment]
> What's the complaint? That the Raw File view isn't useful and takes up space? I'm curious how many users are voicing complaints and what they are.

The main usecase I observed was people in the Siivagunner group, they mostly had no reason to use the hexviewer and I got asked about it very often. They often expressed requests for a better UX in navigating the collections panel. As of now I'm in the same boat as you as I haven't interacted with the group in a while.

> To use your example of exporting all samples, I was considering that we add back in a column for file type (sequence, sample collection, instrument set), even if we just use icons to preserve space. To export all samples across multiple files, you just sort by type, easily select all sample collections, and right click

This is reasonable

> In light of this, we might consider making the collections/log views movable via a title bar (perhaps a vertical one?).

I have tried vertical bars in the past and they looked wrong on all platforms (Qt 5.9), you are welcome to try again if you'd like.
 
> However, I believe that this PR is a major improvement over the status quo.

I tested it and it feels easier to navigate indeed


Btw, this:
> remove an odd call to setBackgroundRole(QPalette::Dark) after dropping a file into the app

was there to fix a Qt bug where the hexview area would randomly change color after dropping a file

**Review by sykhro — APPROVED** [review]

File: src/ui/qt/MainWindow.cpp [hunk 1]
```diff
@@ -86,24 +89,38 @@ void MainWindow::createElements() {
...
   QList<QDockWidget *> docks = findChildren<QDockWidget *>(QString(), Qt::FindDirectChildrenOnly);
   m_menu_bar = new MenuBar(this, docks);
   setMenuBar(m_menu_bar);
 }
 
+void MainWindow::showEvent(QShowEvent* event) {
+  QMainWindow::showEvent(event);
+
+  // Set the initial heights of the docks in relation to the main window height
+  QList<int> sizes;
```
[the remark this answers is not in the document]
Comment (mikelow) [reply src/ui/qt/MainWindow.cpp:114]: resizeDocks(), called a few lines down, expects a QList:

```
void resizeDocks(const QList<QDockWidget *> &docks,
                     const QList<int> &sizes, Qt::Orientation orientation);
```

**Comment by mikelow** [comment]
> The main usecase I observed was people in the Siivagunner group, they mostly had no reason to use the hexviewer and I got asked about it very often. They often expressed requests for a better UX in navigating the collections panel. As of now I'm in the same boat as you as I haven't interacted with the group in a while.

Ok, that's interesting. We probably should allow batch conversion via the collection list view.

> I have tried vertical bars in the past and they looked wrong on all platforms (Qt 5.9), you are welcome to try again if you'd like.

I just looked further into this and I don't even see a built-in option to allow vertical title bars on dock widgets. Am I missing something?

> I tested it and it feels easier to navigate indeed

Awesome!

> was there to fix a Qt bug where the hexview area would randomly change color after dropping a file

Ah ok. Unfortunately, it's affecting the the dock widget title bar text color. I'm guessing the original issue is moot now that hex view is completely different (and I haven't seen this bug).

**Pull request merged**